crates/runtime/src/memory.rs

```diff
@@ -484,6 +488,11 @@ impl SharedMemory {
     pub fn vmmemory_ptr(&self) -> *const VMMemoryDefinition {
         &self.0.read().unwrap().def.0 as *const _
     }
+
+    /// Return a reference to the shared memory's [ParkingSpot].
+    pub fn parking_spot(&self) -> Arc<ParkingSpot> {
+        self.1.clone()
+    }
```

Comment:
Could this perhaps return `&ParkingSpot` to avoid the need for cloning? I think that should work out for the definitions in the libcalls.

Reply:
err.. did not succeed